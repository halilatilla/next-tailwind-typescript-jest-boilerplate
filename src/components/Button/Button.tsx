import { FC } from 'react'

interface Props {
  label: string
}

const Button: FC<Props> = ({ label }) => {
  return (
    <button className="font-bold text-gray-800" type="button">
      {label}
    </button>
  )
}

export default Button
