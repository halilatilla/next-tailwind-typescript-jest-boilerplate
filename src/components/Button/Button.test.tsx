import { render, screen } from '@testing-library/react'

import Button from './Button'

// Just a sample test.
describe('Index Page', () => {
  it('renders correctly', () => {
    const { container } = render(<Button label="Hello" />)
    expect(container).toMatchSnapshot()
  })

  it('renders button', () => {
    // Given
    render(<Button label="Hello" />)

    // When
    const mainButton = screen.getByRole('button')

    // Then
    expect(mainButton).toBeInTheDocument()
  })
})
