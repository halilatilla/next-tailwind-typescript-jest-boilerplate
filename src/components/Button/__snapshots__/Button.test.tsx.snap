// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Page renders correctly 1`] = `
<div>
  <button
    class="font-bold text-gray-800"
    type="button"
  >
    Hello
  </button>
</div>
`;
